### ISSUE TITLE
`type_contents_order` false positive with `willMove(toParent:)`

### ISSUE THREAD

**Issue by kacper1703** [realm/SwiftLint#3478]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

For a following code in a `UIViewController`:
```swift
override func viewDidDisappear(_ animated: Bool) {
    super.viewDidDisappear(animated)
    viewModel.viewDidDisappear()
}

override func willMove(toParent parent: UIViewController?) { 
    super.willMove(toParent: parent)
    if parent == nil { //detect "Back" tapping
        viewModel.willMoveToParent()
    }
}

override func viewDidLoad() {
    super.viewDidLoad()
    configureUI(with: viewModel)
}
```
the `type_contents_order` is triggered in `willMove(toParent:)`. This is a lifecycle method, so the rule should not be triggered.

### Environment

* SwiftLint version **0.42.0**
* Installation method used - **CocoaPods**
* Xcode 12.2 (12B45b)
* Configuration file:

```yml
opt_in_rules:
- type_contents_order

type_contents_order:
    excluded: ".*Tests\\.swift"
```

**Comment by kacper1703** [comment]
Bumping the issue up :)
